api/rs/slint/docs.rs

```diff
@@ -83,7 +83,7 @@ pub mod generated_code {
         /// ```
         pub fn on_hello(&self, f: impl Fn() + 'static) {}
 
-        /// For each public function declared at the root of the component, a function to call
+        /// For each public function declared at the root of the component, a function to synchronously call
```

Comment:
I’m not sure this adds much value. I mean, this is a Rust function declaration, it returns a value, and it’s not declared async. How could it be interpreted as asynchronous?

I’m not opposed to the patch, but I’d like to better understand how this became unclear. Should this perhaps be documented elsewhere, in the general Slint docs? (It applies to all languages)

Reply:
It's functions to interface with the Slint context, which is kind of a dark pit without deeper knowledge. Okay, `invoke_...()` for `function`s returns a value; but not `invoke_...()` for callbacks. You could argue that something like `changed`-callbacks are also deferred. I just found myself in the situation of not trusting how exactly it behaves, when needing exact information to be able to prevent `RefCell` borrowing panics. And the docs didn't provide a guarantee in this regard. (See also #5885.)

Reply:
But even regarding `function`s: I had `function` that takes a `string` and returns nothing. Because of that, I went directly to the docs text, ignoring the `invoke_...()` signature of the sample component, because I deemed it irrelevant and didn't think of other possible signatures.

Reply:
> it returns a value, and it’s not declared async. How could it be interpreted as asynchronous?

It doesn't always return a value and people wouldn't interpret it as `async` (rust term) but maybe as queued.

I think this clarification is accurate so I'll merge this patch. Thanks